src/modules/job-info/job_state.c

```diff
@@ -1745,45 +1677,126 @@ static int job_events_annotations (struct job_state_ctx *jsctx,
                       __FUNCTION__);
-            goto error;
+            return -1;
         }
 
-        if (!strcmp (name, "priority")) {
-            if (job_events_priority (jsctx, id, timestamp, context) < 0)
-                goto error;
+        if (!strcmp (name, "submit")) {
+            if (job_events_submit (jsctx,
+                                   id,
+                                   eventlog_seq,
+                                   timestamp,
+                                   context) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "depend")) {
+            if (job_events_depend (jsctx,
+                                   id,
+                                   eventlog_seq,
+                                   timestamp) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "alloc")) {
+            if (job_events_alloc (jsctx,
+                                  id,
+                                  eventlog_seq,
+                                  timestamp) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "finish")) {
+            if (job_events_finish (jsctx,
+                                   id,
+                                   eventlog_seq,
+                                   timestamp,
+                                   context) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "clean")) {
+            if (job_events_clean (jsctx,
+                                  id,
+                                  eventlog_seq,
+                                  timestamp) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "priority")) {
+            if (job_events_priority (jsctx,
+                                     id,
+                                     eventlog_seq,
+                                     timestamp,
+                                     context) < 0)
+                return -1;
+        }
+        else if (!strcmp (name, "exception")) {
+            if (job_events_exception (jsctx,
+                                      id,
+                                      eventlog_seq,
+                                      timestamp,
+                                      context) < 0)
+                return -1;
         }
         else if (!strcmp (name, "annotations")) {
             if (job_events_annotations (jsctx, id, context) < 0)
-                goto error;
+                return -1;
         }
+        else {
+            struct job *job;
+            if (!(job = zhashx_lookup (jsctx->index, &id))) {
+                flux_log_error (jsctx->h, "%s: job %ju not in hash",
+                                __FUNCTION__, (uintmax_t)id);
+                return -1;
+            }
+            (void) job_events_update_eventlog_seq (jsctx, job, eventlog_seq);
+        }
+    }
+
+    return 0;
+}
+
+static void job_events_continuation (flux_future_t *f, void *arg)
```

Comment:
job_events_continuation() is a continuation of the function `job_state_create()` which is located far away in the source.  Maybe at least rename that to `job_state_create_continuation()` and relocate adjacent to `job_state_create()`?